pkg/metrics/metrics.go

```diff
@@ -235,6 +265,19 @@ func (c *Counter) IsMetricNodeUpgradeEndTimeSet(upgradeConfigName string, versio
 	return false, nil
 }
 
+func (c *Counter) IsSilenceActive() (bool, error) {
```

Comment:
Could you also check that the silence is created by the managed-upgrade-operator?
See -> https://github.com/openshift/managed-upgrade-operator/blob/ff4fbb232623e0609b10f748adc1bb73fcc0891c/pkg/maintenance/alertmanagerMaintenance.go#L166